packages/react-ui-components/src/components/CodeEditor/CodeEditor.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useMemo, useRef } from 'react';
+import Editor, { OnMount, useMonaco } from '@monaco-editor/react';
+import type { editor } from 'monaco-editor';
+import { DEFAULT_CHANGE_DEBOUNCE_TIME, DEFAULT_EDITOR_MODE, DEFAULT_EDITOR_THEME, DEFAULT_MONACO_OPTIONS } from './config';
+import { ECodeEditorTheme, ICodeEditorProps, OnEditorChangeCallback } from './types';
+import { DebouncedFunc, debounce } from 'lodash-es';
+import { KvLoader } from '../stencil-generated';
+
+export const KvCodeEditor = ({
+	code,
+	readOnly = false,
+	debounceTime = DEFAULT_CHANGE_DEBOUNCE_TIME,
+	loadingComponent = <KvLoader />,
+	mode = DEFAULT_EDITOR_MODE,
+	theme = DEFAULT_EDITOR_THEME,
+	customTheme,
+	onChange
+}: ICodeEditorProps) => {
+	const monaco = useMonaco();
+	const editorRef = useRef<editor.IStandaloneCodeEditor>();
+	const readOnlyRef = useRef<boolean>(readOnly);
+
+	const onChangeDebounced: DebouncedFunc<OnEditorChangeCallback> = useMemo(() => debounce(value => onChange?.(value), debounceTime), []);
+	const onEditorMount: OnMount = useMemo(() => editor => (editorRef.current = editor), []);
+
+	useEffect(() => {
+		if (!monaco) return;
+		if (customTheme) {
+			monaco.editor.defineTheme(ECodeEditorTheme.Custom, customTheme);
+			monaco.editor.setTheme(ECodeEditorTheme.Custom);
+		}
+	}, [monaco, customTheme]);
+
+	useEffect(() => {
+		if (readOnlyRef?.current !== readOnly) {
+			readOnlyRef.current = readOnly;
+		}
+		editorRef.current?.updateOptions({ readOnly });
+	}, [editorRef.current, readOnly]);
```

Comment:
Does this readonly control method work? Could not check via storybook. Why did you choose to follow this approach instead of using the options prop from the Editor?

Reply:
This is a tricky one, the monaco editor allows you to pass the readOnly prop in it's constructor options, but then to update it you must use the `updateOptions` function, this wrapper gives us a few [options](https://www.npmjs.com/package/@monaco-editor/react#editor-instance) to retrieve the editor's instance after rendering it, and this approach seemed like the most logical to me when it comes to preventing unnecessary re-renders.

Regarding the storybook responsiveness, I'm not sure what you meant, the control appears to work fine, is there something I'm missing here?

Let me know what you think :)

Reply:
Yep sounds good! The read only prop was not working on my side but it seems its ok, maybe I was being 👜  😄